### PR TITLE
Try TTS in pipeline form

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

Add "Try TTS" button.
Messages are saved in the local storage per language. A default one is displayed if the language match the language interface.

CleanShot 2023-04-24 at 16 45 27

CleanShot 2023-04-24 at 16 45 38

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/translations/en.json [hunk 1]
```diff
@@ -1330,6 +1330,13 @@
       "enter_code": {
         "title": "Enter code",
         "input_label": "Code"
+      },
+      "tts-try": {
+        "header": "Try Text to Speech",
+        "message": "Message",
+        "message_example": "Hello. How can I assist?",
+        "message_placeholder": "Enter a sentence to read.",
+        "read": "Read"
```
Comment (balloob) [inline src/translations/en.json:1339]: I would lean to use the world "Play"

Comment (piitaya) [reply src/translations/en.json:1339]: We can also use "Speak"

File: src/dialogs/tts-try/dialog-tts-try.ts [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+                class="loading"
+              ></ha-circular-progress>
+            `
+          : html`
+              <ha-button
+                ?dialogInitialFocus=${Boolean(this._defaultMessage)}
+                slot="primaryAction"
+                .label=${this.hass.localize("ui.dialogs.tts-try.play")}
+                @click=${this._playExample}
+                .disabled=${this._loadingExample || !this._valid}
```
Comment (bramkragten) [inline src/dialogs/tts-try/dialog-tts-try.ts:98]:
```suggestion
                .disabled=${!this._valid}
```

File: src/dialogs/tts-try/dialog-tts-try.ts [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+      return;
+    }
+
+    const platform = this._params!.engine;
+    const language = this._params!.language;
+    const voice = this._params!.voice;
+
+    this._messages = {
+      ...this._messages,
+      [language?.substring(0, 2) || "_"]: message,
```
Comment (bramkragten) [inline src/dialogs/tts-try/dialog-tts-try.ts:126]: Why the `_`? We never retrieve this message?

**Review by bramkragten — DISMISSED** [review]

File: src/panels/config/voice-assistants/assist-pipeline-detail/assist-pipeline-detail-tts.ts [hunk 4]
```diff
@@ -38,8 +40,6 @@ export class AssistPipelineDetailTTS extends LitElement {
                   },
                 }
               : { name: "", type: "constant" },
-
-            { name: "", type: "constant" },
```
[the remark this answers is not in the document]
Comment (piitaya) [reply src/panels/config/voice-assistants/assist-pipeline-detail/assist-pipeline-detail-tts.ts:42]: The Try TTS button is on the left so It's weird to have space now 😅

**Review by bramkragten — APPROVED** [review]

**Pull request merged**